utils/babeltrace_thapi.in

```diff
@@ -273,10 +273,28 @@ def modify_metadata(command, trace)
   File.write(File.join($options[:output], THAPI_METADATA_FILE), y.to_yaml)
 end
 
-def create_and_run_graph(command, inputs)
-  def no_gc(command, inputs)
+def bt_graphs
+  source_comp = if $options[:live]
+                  'source.ctf.lttng_live'
+                else
+                  'source.ctf.fs'
+                end
+
+  { 'tally' => ['filter.utils.muxer', 'filter.intervals.interval',
+                'filter.btx_aggreg.aggreg', 'sink.btx_tally.tally'],
+    'timeline' => ['filter.utils.muxer', 'filter.intervals.interval',
+                   'sink.btx_timeline.timeline'],
+    'trace' => ['filter.utils.muxer', 'sink.text.rubypretty'],
+    'to_interval' => ['filter.utils.muxer', 'filter.intervals.interval',
+                      'filter.btx_stripper.stripper', 'sink.ctf.fs'],
+    'to_aggreg' => ['filter.utils.muxer', 'filter.intervals.interval', 'filter.btx_aggreg.aggreg',
+                    'filter.btx_stripper.stripper', 'sink.ctf.fs'] }.transform_values { |l| [source_comp] + l }
+end
```

Comment:
```suggestion
$thapi_graphs = { 'tally' => ['source.ctf.fs', 'filter.utils.muxer', 'filter.intervals.interval',
                             'filter.btx_aggreg.aggreg', 'sink.btx_tally.tally'],
                 'timeline' => ['source.ctf.fs', 'filter.utils.muxer', 'filter.intervals.interval',
                                'sink.btx_timeline.timeline'],
                 'trace' => ['source.ctf.fs', 'filter.utils.muxer', 'sink.text.rubypretty'],
                 'trace_live' => ['source.ctf.lttng_live', 'filter.utils.muxer', 'sink.text.rubypretty'],
                 'to_interval' => ['source.ctf.fs', 'filter.utils.muxer', 'filter.intervals.interval',
                                   'filter.btx_stripper.stripper', 'sink.ctf.fs'],
                 'to_aggreg' => ['source.ctf.fs', 'filter.utils.muxer', 'filter.intervals.interval',
                                 'filter.btx_aggreg.aggreg', 'filter.btx_stripper.stripper', 'sink.ctf.fs'] }

def bt_graph(command)
  source_comp = if $options[:live]
                  'source.ctf.lttng_live'
                else
                  'source.ctf.fs'
                end
  [source_comp] + $thapi_graphs[command]
end
```

What do you think of this?

Reply:
I don't like the function instantiating a big data structure.

Reply:
or make `$thapi_graphs` a constant `THAPI_GRAPHS`.

Reply:
I don't like have `THAPI_GRAPHS` as a variable in global scope. As it should't not be used, it need to be modified by `bt_graph` to be "complete"  and usable.